PR: https://github.com/odoo/odoo/pull/

From: db02f776cb34ea49f4f59391b5bf414b7b76b3b9
From: Xavier Morel
Date: 2021-04-29 05:34:20

Breaking data model changes score: 3, change matches:
-    custom_code_head = fields.Text('Custom <head> code')
-    custom_code_footer = fields.Text('Custom end of <body> code')
-    robots_txt = fields.Text('Robots.txt', translate=False, groups='website.group_website_designer')

Total Changes: 6

[FIX] website: de-raw-ify

Convert `custom_code_*` to `Html` with no sanitization, such that qweb
considers them markup-safe.

`base_popup_options` value comes from a rendered XML body so should
already be safe

raw -> out in the extended website info template, not sure why this
was even raw

Also mark robots_txt as an unsanitized Html field, that's
not *actually* true but it is true enough for our purposes, especially
since the field gets edited through a text-mode wizard.

================================= pseudo patch: =================================

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -117,10 +117,10 @@ class Website(models.Model):
     partner_id = fields.Many2one(related='user_id.partner_id', string='Public Partner', readonly=False)
     menu_id = fields.Many2one('website.menu', compute='_compute_menu', string='Main Menu')
     homepage_id = fields.Many2one('website.page', string='Homepage')
-    custom_code_head = fields.Text('Custom <head> code')
-    custom_code_footer = fields.Text('Custom end of <body> code')
+    custom_code_head = fields.Html('Custom <head> code', sanitize=False)
+    custom_code_footer = fields.Html('Custom end of <body> code', sanitize=False)
 
-    robots_txt = fields.Text('Robots.txt', translate=False, groups='website.group_website_designer')
+    robots_txt = fields.Html('Robots.txt', translate=False, groups='website.group_website_designer', sanitize=False)
 
     def _default_favicon(self):
         img_path = get_resource_path('web', 'static/src/img/favicon.ico')
